nomos-mix/core/src/processor/mod.rs

```diff
@@ -0,0 +1,72 @@
+mod crypto;
+mod temporal;
+
+pub use crypto::CryptographicProcessorSettings;
+pub use temporal::TemporalProcessorSettings;
+
+use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc;
+
+use crate::processor::crypto::CryptographicProcessor;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct ProcessorSettings {
+    pub cryptographic_processor: CryptographicProcessorSettings,
+    pub temporal_processor: TemporalProcessorSettings,
+}
+
+/// [`Processor`] implements "Message Router" defined in the Tier-2 spec.
+/// - Wraps new messages using [`CryptographicProcessor`]
+/// - Unwraps incoming messages received from network using [`CryptographicProcessor`]
+/// - Pushes unwrapped messages to [`TemporalProcessor`] to delay them
+/// - Releases messages returned by [`TemporalProcessor`]
+pub struct Processor {
+    /// To receive new messages originated from this node
+    new_message_receiver: mpsc::UnboundedReceiver<Vec<u8>>,
+    /// To receive incoming messages from received the network
```

Comment:
"To receive incoming messages from the network"?

Reply:
Thanks. It's a leftover. Fixed in 7091a5ee2210c45d8674f8a3b76bd005440f4e3f